File: dcp-solutions/2024-10-18/solution.go
```markdown
# Conway's Game of Life in Golang

### game_of_life.go

```go
package main

import (
    "fmt"
    "math/bits"
)

type Cell struct {
    Row   int
    Col   int
    Alive bool
}

type Board struct {
    Rows, Cols int
    Cells     []Cell
}

func NewBoard(rows, cols int) *Board {
    return &Board{
        Rows: rows,
        Cols: cols,
        Cells: make([]Cell, rows*cols),
    }
}

func (b *Board) InitializeCells(liveCoordinates [][]int) {
    for _, coordinate := range liveCoordinates {
        row := coordinate[0] % b.Rows
        col := coordinate[1] % b.Cols
        b.Cells[row*b.Cols+col].Alive = true
    }
}

func (b *Board) NextGeneration() *Board {
    newBoard := NewBoard(b_ROWS, b_COLS)
    for row := 0; row < b_ROWS; row++ {
        for col := 0; col < b_COLS; col++ {
            liveNeighbors := b.GetLiveNeighbors(row, col)
            if b.Cells[row*b.Cols+col].Alive && (liveNeighbors < 2 || liveNeighbors > 3) {
                newBoard.Cells[newBoard.Rows*newBoard.Cols+row*newBoard.Cols+col].Alive = false
            } else if !b.Cells[row*b.Cols+col].Alive && liveNeighbors == 3 {
                newBoard.Cells[newBoard.Rows*newBoard.Cols+row*newBoard.Cols+col].Alive = true
            }
        }
    }
    return newBoard
}

func (b *Board) Print() {
    for row := 0; row < b_ROWS; row++ {
        for col := 0; col < b_COLS; col++ {
            if b.Cells[row*b.Cols+col].Alive {
                fmt.Print("*")
            } else {
                fmt.Print(".")
            }
        }
        fmt.Println()
    }
}

const (
    b_ROWS = 100
    b_COLS = 100
)

func main() {
    board := NewBoard(b_ROWS, b_COLS)
    liveCoordinates := [][]int{
        {0, 0},
        {1, 1},
        {2, 2},
        // Add more coordinates here...
    }
    board.InitializeCells(liveCoordinates)

    for step := 0; step < 100; step++ {
        fmt.Println("Step:", step+1)
        board.Print()
        board = board.NextGeneration()
    }
}
```

File: dcp-solutions/2024-10-18/solution.py
```python
class GameOfLife:
    def __init__(self, live_cells, steps):
        self.live_cells = set(live_cells)
        self.steps = steps

    def next_generation(self):
        new_board = set()
        for x in range(-10, 11): # assume the board is at least 20x20
            for y in range(-10, 11): 
                live_neighbours = 0
                for dx in [-1, 0, 1]:
                    for dy in [-1, 0, 1]: 
                        nx, ny = x + dx, y + dy
                        if (nx == 0 and ny == 0) or (dx != 0 or dy != 0):
                            continue
                        if (nx, ny) in self.live_cells:
                            live_neighbours += 1 
                if nx == 0 and ny == 0: 
                    continue 
                state = 0 # dead cell initially
                if (nx, ny) in self.live_cells or live_neighbours in [2, 3]:
                    state = 1 # alive cell
                new_board.add((nx, ny)) if state else new_board.discard((nx, ny))
        self.live_cells = new_board

    def play(self):
        for i in range(self.steps): 
            print("Step: ", i + 1)
            board_str = ""
            for x in range(-10, 11): 
                for y in range(-10, 11): 
                    if (x, y) in self.live_cells:
                        board_str += "* "
                    else:
                        board_str += ". "
                print(board_str)
            print("\n")
            self.next_generation()
```